### PR TITLE
remove font_assets gem as it fails with rails 5.1

### PR DESCRIPTION
**Pull request by akostadinov**
According to docs it only helps cases where assets are *not*
pre-compiled. Gem is presently only included for production and preview
environments where assets *are* pre-compiled.

For a reference, gem was introduced with (bb04f2f8b82 and 8708422380c).

**What this PR does / why we need it**:

remove obsolete dependency

**Which issue(s) this PR fixes** 

part of https://issues.redhat.com/browse/THREESCALE-5010

### CODE REVIEW

File: config/initializers/assets.rb [hunk 1]
```diff
@@ -8,5 +8,32 @@
...
+  private
+
+  def access_control_headers
+    Rails.configuration.public_file_server.headers
+  end
+end
+
+Rails.application.configure do
+  if defined?(ActionDispatch::Static)
+    config.middleware.insert_before ActionDispatch::Static, AssetsOptionsResponder
```
Comment (akostadinov) [inline config/initializers/assets.rb:37]: We will need to handle `OPTIONS` requests by ourselves. This is probably marginally better than `font_assets` gem which also handles normal respond headers and we just setup in the public server.

Should we go back to `font_assets`? Or with custom code like the above?

btw the above code needs to respond only to requests served by public file server. I'm not sure how to check the path yet. In worst case I can do like `font_assets` and check for file extension. But I don't like this approach a lot.

**Comment by akostadinov** [comment]
Before patch:
```
HTTP/1.1 200 OK
Last-Modified: Wed, 01 Sep 2021 08:28:04 GMT
Content-Type: application/font-woff
Cache-Control: public, max-age=31557600
Content-Length: 36749
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET
Access-Control-Allow-Headers: x-requested-with
Access-Control-Max-Age: 3628800
Vary: Accept-Encoding
X-Frame-Options: DENY
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Connection: keep-alive
Server: thin

🐚 curl -v -X OPTIONS http://provider-admin.3scale.localhost:3000/packs/media/RedHatDisplay/RedHatDisplay-Medium-3efbce15.woff
*   Trying ::1:3000...
* connect to ::1 port 3000 failed: Connection refused
*   Trying 127.0.0.1:3000...
* Connected to provider-admin.3scale.localhost (127.0.0.1) port 3000 (#0)
> OPTIONS /packs/media/RedHatDisplay/RedHatDisplay-Medium-3efbce15.woff HTTP/1.1
> Host: provider-admin.3scale.localhost:3000
> User-Agent: curl/7.76.1
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET
< Access-Control-Allow-Headers: x-requested-with
< Access-Control-Max-Age: 3628800
< Vary: Accept-Encoding
< X-Frame-Options: DENY
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Connection: close
< Server: thin
< 
* Closing connection 0
```

After patch:
```
GET /packs/media/RedHatDisplay/RedHatDisplay-Medium-3efbce15.woff HTTP/1.1
Host: provider-admin.3scale.localhost:3000
User-Agent: Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:89.0) Gecko/20100101 Firefox/89.0
Accept: application/font-woff2;q=1.0,application/font-woff;q=0.9,*/*;q=0.8
Accept-Language: en-US,en;q=0.7,bg;q=0.3
Accept-Encoding: identity
Connection: keep-alive
Referer: http://provider-admin.3scale.localhost:3000/p/login
Cookie: _system_session=VUVzWVRVenpZZmlOM0M0UVIvT2dtdC9HRUgrejZJejUzNXQrMGxnM1A3NzJDVzFUY0pQTHBjcDNHbWo0bjFRcTI4RGl2UXR3ZUZheEFzMXNQS29ycHBBU0VzWC9Ec2JBMC91d0V4TTN6MlZvVnBrNXpGbitiV0RyNk1TWmhqclhYNjFFZUVsVTI4emVXSGFRMVhXMm50MkMrdkl0cytZS29WNm9zWmx6N01xaTE2OEdUOUpVZ1BIbXovRDhSSWNVLS1tb0hNTTJvNmdabGpCLzRZQ0ZOYzZBPT0%3D--5930c7c0fdfd543f0ff5dc7372b7ece12021b497; user_session=Ill3YmtaWGRYV1RSS2k0Nk9YZkY0UThpMTNyZl9GY0liSmUteWsyal9GYlki--e1e10d173381f7e69b5c64e202650481e7335077
Pragma: no-cache
Cache-Control: no-cache

HTTP/1.1 200 OK
Last-Modified: Wed, 01 Sep 2021 08:28:04 GMT
Content-Type: application/font-woff
Cache-Control: public, max-age=31557600
Access-Control-Allow-Origin: *
Access-Control-Allow-Headers: x-requested-with
Access-Control-Allow-Methods: GET
Access-Control-Max-Age: 3628800
Content-Length: 36749
Vary: Accept-Encoding
X-Frame-Options: DENY
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Connection: keep-alive
Server: thin


🐚 curl -v -X OPTIONS http://provider-admin.3scale.localhost:3000/
*   Trying ::1:3000...
* connect to ::1 port 3000 failed: Connection refused
*   Trying 127.0.0.1:3000...
* Connected to provider-admin.3scale.localhost (127.0.0.1) port 3000 (#0)
> OPTIONS / HTTP/1.1
> Host: provider-admin.3scale.localhost:3000
> User-Agent: curl/7.76.1
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 204 No Content
< Cache-Control: public, max-age=31557600
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Headers: x-requested-with
< Access-Control-Allow-Methods: GET
< Access-Control-Max-Age: 3628800
< X-Frame-Options: DENY
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Connection: close
< Server: thin
< 
* Closing connection 0
```

**Comment by hallelujah** [comment]
I am holding that until we release 2.11

**Comment by akostadinov** [comment]
superseded by #2602

**Pull request closed without merging**